### ISSUE TITLE
Prediction Code for weights created by ex_dnn_training.py throws exception

### ISSUE THREAD

**Issue by Benuraj** [mpatacchiola/deepgaze#6]
Hi I'm new to TF, while trying to plug-in the weights created by ex_dnn_training.py in ex_cnn_head_pose_estimation_images_pitch.py i get following error
Tensor name "out_pitch_w" not found in checkpoint files ../dnn_1600i_ben_4h_3o-50000

Is the Model / Hyperparameter for given weight cnn_cccdd_30k.tf and that defined in ex_dnn_training same?

**Comment by mpatacchiola** [comment]
Hi Benuraj

The code inside ex_dnn_training.py is for training a deep neural network which is a multilayer perceptron. The code in the ex_cnn_head_pose_estimation_images_pitch.py uses a Convolutional Neural Network.
The two models are then different and you cannot use configuration files of the former to run the latter. Unfortunately I still cannot share many details of the work I did on CNN because I have an article under review. What I can do is to expand the code for training the deep neural network adding a file for using the model. In the next days I'm gonna do it. Stay tuned...

**Comment by Benuraj** [comment]
okay it would be great if test file for MLP is added. Thanks

**Comment by Benuraj** [comment]
You had time to look into this?

**Comment by mpatacchiola** [comment]
Hi Benuraj

I am going to work on it from Tuesday when I will be back to home. The last weeks I have been travelling. I will share the code as soon as I can.

**Comment by Benuraj** [comment]
okay thanks

**Comment by mpatacchiola** [comment]
Hi Benuraj

I finalised the code for the DNN and I added a file for training a new model and a file for testing the model after the training. You can find the examples in the folder [ex_dnn_head_pose_estimation_training](https://github.com/mpatacchiola/deepgaze/tree/master/examples/ex_dnn_head_pose_estimation_training). There are also two files for creating pickle files from the Prima head pose dataset and the AFLW dataset. If you follow this procedure you should be able to train your classifier:

1. First of all generate your own dataset, using the prima and AFLW file. Inside the file you can find the link to the images.

2. When you have your dataset preprocessed (it should be a pickle file containing the images and the labels) then you can run the training file. At the end of the training you obtain a tensorflow file which contains the weights of the network.

3. In the last phase you can use the test file to load your weights in a new model and give as input an image to check.

I suggest you to look to the code and change different parameters. Moreover I did not have time to check if everything works fine, it is possible that there are some typos and small bugs around.

All the best, Massimiliano

**Comment by aman-captech** [comment]
@mpatacchiola you have mentioned a checkpoint file in the ex_dnn_head_pose_estimation_testing.py: "dnn_1600i_4h_3o"  but I was unable to find the same in your repo. It would be really helpful if you could provide the ckpt file for the same since I don't have a good dataset to train the model.

**Comment by mpatacchiola** [comment]
Hi @aman-captech 
I should have a few checkpoints but they are quite heavy and I cannot upload it in the repo. If you give me your email address in a private message I can share it with you.
